### PR TITLE
Rewriting bunch of stuff

### PR DESCRIPTION
**Pull request by Mr-Auto**
Putting PR out to see the changes better, maybe get feedback while working on this
#66

### CODE REVIEW

File: src/QtHelpers/TreeViewMemoryFields.cpp [hunk 1]
```diff
@@ -50,10 +50,10 @@ S2Plugin::TreeViewMemoryFields::TreeViewMemoryFields(ViewToolbar* toolbar, Memor
...
 }
 
-void S2Plugin::TreeViewMemoryFields::setMemoryMappedData(MemoryMappedData* mmd)
-{
-    mMemoryMappedData = mmd;
-}
+//void S2Plugin::TreeViewMemoryFields::setMemoryMappedData(MemoryMappedData* mmd)
+//{
+//    mMemoryMappedData = mmd;
+//}
```
Comment (Mr-Auto) [inline src/QtHelpers/TreeViewMemoryFields.cpp]: This whole thing with `MemoryMappedData` here seams weird, only ever used for `VirtualFunctionTable`. Will look for a fix later

File: include/Configuration.h [hunk 2]
```diff
@@ -1,64 +1,298 @@
...
+    constexpr uint8_t gsColMemoryOffset = 5;
+    constexpr uint8_t gsColMemoryOffsetDelta = 6;
+    constexpr uint8_t gsColType = 7;
+    constexpr uint8_t gsColComment = 8;
+
+    /*constexpr uint16_t gsRoleField = Qt::UserRole + gsColField;
+    constexpr uint16_t gsRoleValue = Qt::UserRole + gsColValue;
+    constexpr uint16_t gsRoleValueHex = Qt::UserRole + gsColValueHex;
+    constexpr uint16_t gsRoleComparisonValue = Qt::UserRole + gsColComparisonValue;
+    constexpr uint16_t gsRoleComparisonValueHex = Qt::UserRole + gsColComparisonValueHex;*/
```
Comment (Mr-Auto) [inline include/Configuration.h]: All unused, not sure if there was some plan/intention for those

Comment (zappatic) [reply include/Configuration.h]: Probably were used at some point to store the comparison values, not sure why they're not anymore.

File: src/Spelunky2.cpp [hunk 3]
```diff
@@ -1,121 +1,110 @@
...
+    if (ptr != nullptr)
     {
-        findSpelunky2InMemory();
+        delete ptr;
+        ptr = nullptr;
     }
-    return gSpelunky2AfterBundleSize;
 }
 
 // size_t S2Plugin::Spelunky2::findEntityListMapOffset()
```
Comment (Mr-Auto) [inline src/Spelunky2.cpp:103]: Wait, this commented out code here
```cpp
auto mapPtr = reinterpret_cast<std::unordered_map<std::string, uint16_t>*>(mapOffset);
dprintf("mapPtr size = %d\n", mapPtr->size());
```
I thought that it's impossible to read game memory without using `Script::Memory::Read*` functions?

File: src/QtHelpers/TreeViewMemoryFields.cpp [hunk 4]
```diff
@@ -371,1056 +380,1170 @@ void S2Plugin::TreeViewMemoryFields::updateTableHeader(bool restoreColumnWidths)
...
-            auto comparisonStateTitle = QString::fromStdString(std::to_string(comparisonValue) + ": " + mToolbar->configuration()->stateTitle(stateFieldName, comparisonValue));
-            itemComparisonValue->setData(comparisonStateTitle, Qt::DisplayRole);
-            itemComparisonValue->setBackground(value != comparisonValue ? comparisonDifferenceColor : Qt::transparent);
-            itemComparisonValueHex->setBackground(value != comparisonValue ? comparisonDifferenceColor : Qt::transparent);
+            auto value = getDataFrom(itemField->parent()->index(), gsColValue, gsRoleRawValue).toUInt();
+            auto flagSet = ((value & mask) == mask);
+            auto flagTitle = QString::fromStdString(flagName.empty() ? Configuration::get()->flagTitle("unknown", flagIndex) : flagName); // TODO: don't show empty unless it was chosen in settings
+            // TODO: would love to instead get the names and save them in addMemoryField and then just use itemValue->setForeground or itemValue->setData(Qt::TextColorRole) for the color
+            // but it doesn't work with HTML delagate, and i don't know how to edit it to make it work
```
Comment (Mr-Auto) [inline src/QtHelpers/TreeViewMemoryFields.cpp:938]: Would appreciate some help with this, assuming it is possible

**Comment by zappatic** [comment]
Please pause work on this PR. We are going to reorganize the project. It will be moved to my personal account (before this PR), then forked under the spelunkyfyi organisation. You will also get write access to the fork.
Seeing as probably all issues and PR's will be removed in this transfer, make sure to save all the changes you need, so you can reapply them after the fork has happened.

Let me know when you are ready with your backup.

**Comment by Mr-Auto** [comment]
@zappatic I'm ready

**Comment by Mr-Auto** [comment]
So , I assume I should move this PR to the spelunkyfyi fork now?

Are you cutting your involvement completely now?

**Comment by zappatic** [comment]
Well I had assumed the transfer would have wiped out all the PR's, that's why I asked to back everything up so you wouldn't lose your work. Apparently they are moved over, so now you can indeed reapply it to the fork. I think you could either make a patch and reapply, or maybe point your origin on your local fork to the spelunky-fyi fork, not sure what all is possible here, but shouldn't be that hard.

It's not necessarily cutting involvement, I just wanted you to have write access to the fork, so you can work on it directly. You are free to implement whatever you see fit of course, I just don't have the time to verify these huge commits. Have fun with it ;)

**Pull request closed without merging**